vectors_deep1b/challenges/default.json

```diff
@@ -0,0 +1,55 @@
+{
+  "name": "index-and-query",
+  "description": "indexes vectors, force merges to a single segment, queries",
+  "default": true,
+  "schedule": [
+    {
+      "operation": {
+        "operation-type": "delete-index"
+      }
+    },
+    {
+      "operation": {
+        "operation-type": "create-index"
+      }
+    },
+    {
+      "operation": {
+        "operation-type": "cluster-health",
+        "request-params": {
+          "wait_for_status": "green"
+        },
+        "retry-until-success": true
+      }
+    },
+    {
+      "operation": "index-append",
+      "warmup-time-period": 0,
```

Comment:
Shouldn't we have some warmup time so that indexing results (esp. max / min) don't get skewed unreasonably due to jit warmup? Typically we've used something like 240s (https://github.com/elastic/rally-tracks/blob/fe23f98a77115d7762bf43a66053cc18d9b1decc/nyc_taxis/challenges/default.json#L33), which is usually derived by looking at the metrics store.

Reply:
@dliappis  Thanks for the comment,  happy to add `warmup-time-period`, andI will follow your recommendation. about the correct number. 

Does `240` means that we start collecting `index-append` metrics only after 240s? 
Looking at the Median Throughput of   11000 docs/s,  it means that we start collecting metrics after indexing 2'640'000 docs, or around 25% dataset?   Does this look right to you or should it be smaller?

Also, I have noticed that if I ran `rally` with a smaller ingest_percentage, e.g. `ingest_percentage:10", I may not get any metrics in this case?

Reply:
> Does `240` means that we start collecting `index-append` metrics only after 240s?

@mayya-sharipova Correct. The warmup metrics are also available in the metrics store (`sample-type`: `warmup` vs `normal`).

> Also, I have noticed that if I ran `rally` with a smaller ingest_percentage, e.g. `ingest_percentage:10", I may not get any metrics in this case?

Right and while it may feel inconvenient, it actually protects the user from relying on performance figures that can't be 100% trusted.

A decent workaround is to use variable with a default value e.g.

`"warmup-up-period": {{ bulk_warmup | default(240) | int }}`

and thus if you run in a different env to the nightlies, you can override with `--track-params='bulk_warmup:0'`.

Regarding the actual default value, it's just what's used in other tracks, ideally we can run on the actual nightly hardware and using a simple visualization see if 240s is needed for warmup, or less.

Reply:
@dliappis Thanks for the explanation, it makes sense. I've added modified the warmup period as suggested in 8667d92cca5d56ca3706fc64a8668bf19731aa59